src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint no-param-reassign: "error" */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+type PostsState = {
+  posts: Post[],
+  loaded: boolean,
+  hasError: boolean,
+};
+const initialState: PostsState = {
+  posts: [],
+  loaded: true,
+  hasError: false,
+};
+
+export const getAsyncPosts = createAsyncThunk(
+  'posts/get',
```

Comment:
The name of the async thunk should follow the format `domain/eventName`. Consider renaming it to 'posts/fetchPosts' for better clarity.